coverage/build.gradle

```diff
@@ -17,15 +19,35 @@ tasks.register("jacoco-merge", JacocoReport) {
     classDirectories = files(jprojects.sourceSets.main.output)
     reports {
         html.required = true
+        html.outputLocation = layout.buildDirectory.dir('./reports/java/java-coverage.html')
         csv.required = true
+        csv.destination = file("$buildDir/reports/java/java-coverage.csv")
         xml.required = false
     }
     def projRootDir = rootProject.rootDir.absolutePath
     executionData fileTree(projRootDir).include("**/build/jacoco/*.exec")
 }
 
+tasks.register("python-merge") {
+    def projRootDir = rootProject.rootDir.absolutePath
+    def outputDir = file("$buildDir/reports/python/")
+    def outputFile = new File(outputDir, "python-coverage.tsv")
+    outputDir.mkdirs()
+    outputFile.text = ''
+    fileTree(projRootDir).include("**/build/test-results/**/coverage/python-coverage-*.tsv").each { f ->
+        outputFile << f.text + '\n'
+    }
+    fileTree(projRootDir).include("**/build/test-results/**/coverage/python-coverage-*.html/index.html").each { f ->
+        copy {
+          from f.parentFile
+          into new File(outputDir, f.parentFile.name)
+        }
+    }
```

Comment:
The physical execution of a task should be done as part of a doLast block; otherwise, this sort of execution happens when the task is materialized.